lib/cfg-block.c

```diff
@@ -61,6 +61,50 @@ cfg_block_format_name(CfgBlockGenerator *s, gchar *buf, gsize buf_len)
   return buf;
 }
 
+/* token block args */
+
+static void
+_validate_args_callback(gpointer k, gpointer v, gpointer user_data)
+{
+  CfgArgs *defs = ((gpointer *) user_data)[0];
+  const gchar *reference = ((gpointer *) user_data)[1];
+  gpointer *problem_found = &((gpointer *) user_data)[2];
+
+  if ((!defs || cfg_args_get(defs, k) == NULL))
+    {
+      if (cfg_args_is_accepting_varargs(defs))
+        {
+          msg_verbose("Unknown argument, adding it to __VARARGS__",
+                      evt_tag_str("argument", k),
+                      evt_tag_str("value", v),
+                      evt_tag_str("reference", reference));
+        }
+      else
+        {
+          msg_error("Unknown argument specified to block reference",
+                    evt_tag_str("argument", k),
+                    evt_tag_str("value", v),
+                    evt_tag_str("reference", reference));
+          *problem_found = (gpointer) TRUE;
+        }
+    }
+}
+
+static gboolean
+_validate_args(CfgArgs *self, CfgArgs *defs, const gchar *reference)
+{
+  gpointer validate_params[] = { defs, (gchar *) reference, NULL };
```

Comment:
Please use FALSE instead of NULL.

Reply:
I've changed this code to pass a gboolean* as third parameter so that I can name the returned value, which makes the code easier to read. I'll push this in a moment.